### ISSUE TITLE
[RFE] rke2 cluster provisioning - Audit logging configuration support

### ISSUE THREAD

**Issue by snasovich** [rancher/terraform-provider-rancher2#1194]
Terraform side of https://github.com/rancher/rancher/issues/32784

**Comment by jiaqiluo** [comment]
can be validated in https://github.com/rancher/terraform-provider-rancher2/releases/tag/v4.0.0-rc3 

QA: Please check the example TF config file in the linked PR

**Comment by jiaqiluo** [comment]
Hi @slickwarren,  please assist in arranging the issue in the QA team.

**Comment by markusewalker** [comment]
**QA TEST PLAN**

Scenarios
|Scenario|Test Case|
|---|---|
|1|As a standard user, provision a downstream RKE2 node driver cluster with audit logging enabled, validate that auditing is enabled|
|2|As a standard user, provision a downstream RKE2 node driver cluster, edit the cluster to enable audit logging, validate that auditing is enabled|
|3|As a standard user, provision a downstream K3S node driver cluster with audit logging enabled, validate that auditing is enabled|
|4|As a standard user, provision a downstream K3S node driver cluster, upgrade the cluster w/audit loging enabled, validate that auditing is enabled|

**Comment by markusewalker** [comment]
Validated that this is addressed on `v2.8-head` https://github.com/rancher/rancher/commit/6cd32b985f0b39ef2eed363458f86a352c67a42b. See below:

**ENVIRONMENT DETAILS**
- Rancher install: Docker
- Rancher version: `v2.8-head` https://github.com/rancher/rancher/commit/6cd32b985f0b39ef2eed363458f86a352c67a42b
- TF version: `v4.0.0-rc3`

**TEST RESULT**
|Scenario|Test Case|Result
|---|---|---|
|1|As a standard user, provision a downstream RKE2 node driver cluster with audit logging enabled, validate that auditing is enabled|:white_check_mark:|
|2|As a standard user, provision a downstream RKE2 node driver cluster, edit the cluster to enable audit logging, validate that auditing is enabled|:white_check_mark:|
|3|As a standard user, provision a downstream K3S node driver cluster with audit logging enabled, validate that auditing is enabled|:white_check_mark:|
|4|As a standard user, provision a downstream K3S node driver cluster, upgrade the cluster w/audit loging enabled, validate that auditing is enabled|:white_check_mark:|

**VALIDATION STEPS**

**Scenario 1**
1. In a client machine, ran `terraform apply` with the following blocks specified in my `main.tf`:
```
resource "rancher2_secret_v2" "foo" {
  cluster_id = "local"
  name       = "mew-tu1-secret"
  namespace  = "fleet-default"
  data = {
    audit-policy = "IyBMb2cgYWxsIHJlcXVlc3RzIGF0IHRoZSBNZXRhZGF0YSBsZXZlbC4KYXBpVmVyc2lvbjogYXVkaXQuazhzLmlvL3YxCmtpbmQ6IFBvbGljeQpydWxlczoKLSBsZXZlbDogTWV0YWRhdGE="
  }
  annotations = {
    "rke.cattle.io/object-authorized-for-clusters" = "mew-tf-rke2"
  }
}
```

```

resource "rancher2_cluster_v2" "cluster" {
.
.
.
	machine_selector_files {
	      machine_label_selector {
	        match_labels = {
	          "rke.cattle.io/control-plane-role" = "true"
	        }
	      }
	      file_sources {
	        secret {
	          name                = "mew-tu1-secret"
	          default_permissions = "644"
	          items {
	            key         = "audit-policy"
	            path        = "/etc/rancher/rke2/mew-policy.yaml"
	            permissions = "666"
	          }
	        }
	      }
	}
```
2. Validated in my client machine that the `terraform apply` created all of my specified resources.
3. Navigated to Rancher and validated that the downstream RKE2 `v1.27.6+rke2r1` node driver cluster and nodes are `Active` in Cluster Management and Cluster Explorer.
4. SSH'ed into the control plane node and validated that the YAML is populated:
```
$ pwd
/etc/rancher/rke2
$ ll
total 24
drwxr-xr-x 3 root root 4096 Oct 13 21:35 ./
drwxr-xr-x 5 root root 4096 Oct 13 21:35 ../
drwxr-xr-x 2 root root 4096 Oct 13 21:35 config.yaml.d/
-rw-rw-rw- 1 root root  144 Oct 13 21:35 mew-policy.yaml
-rw------- 1 root root    0 Oct 13 21:35 registries.yaml
-rw------- 1 root root  365 Oct 13 21:35 rke2-pss.yaml
-rw------- 1 root root 2973 Oct 13 21:35 rke2.yaml
```
5. Ran `terraform destroy` and validated that all created resources in my `main.tf` were successfully deleted.

_______

**Scenario 2**
1. Repeated scenario 1, steps 1-3, but did not specify the blocks mentioned above to provision my downstream RKE2 `v1.27.6+rke2r1` cluster.
2. Edited my `main.tf` to now include the blocks specified in scenario 1; applied the changes with `terraform apply`.
3. Validated in my client machine that the `terraform apply` created all of my newly specified resources.
4. Validated that the cluster and nodes remain `Active` after the updates.
5. SSH'ed into the control plane node and validated that the YAML is populated.
6. Ran `terraform destroy` and validated that all created resources in my `main.tf` were successfully deleted.

_______

**Scenario 3**
1. Repeated scenario 1, but provisioned a downstream K3S `v1.27.6+k3s1` instead with my `machine_selector_files.file_sources.path` being `/etc/rancher/k3s/mew-policy.yaml`.
2. Validated that I see the same successful results.

_______

**Scenario 4**
1. Repeated scenario 1, steps 1-3, but did not specify the blocks mentioned above to provision my downstream K3S `v1.26.9+k3s1` cluster.
2. Edited my `main.tf` to update the cluster version to `v1.27.6+k3s1` and to include the resource blocks mentioned in iscenario 1.
3. Validated in my client machine that the `terraform apply` created all of my newly specified resources and upgraded my cluster.
4. Navigated to Rancher and validated that the cluster and nodes have been upgraded to `v1.27.6+k3s1` and are in a state of `Active`.
5. SSH'ed into the control plane node and validated that the YAML is populated.
6. Ran `terraform destroy` and validated that all created resources in my `main.tf` were successfully deleted.
